jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/internal/HttpChannelState.java

```diff
@@ -1574,9 +1574,14 @@ public void succeeded()
                 {
                     httpChannelState._callbackFailure = failure;
                     if (!stream.isCommitted())
+                        // We are not committed, so we can send an error response.
                         errorResponse = new ErrorResponse(request);
-                    else
+                    else if (httpChannelState._handling == null)
```

Comment:
I think this more precisely should be:


```suggestion
                    else if (httpChannelState._handled)
```